synapse/handlers/oidc_handler.py

```diff
@@ -782,8 +789,14 @@ async def handle_redirect_request(
 
         for cookie_name, options in _SESSION_COOKIES:
             request.cookies.append(
-                b"%s=%s; Max-Age=3600; %s"
-                % (cookie_name, cookie.encode("utf-8"), options)
+                b"%s=%s; Max-Age=3600; Path=%s; %s"
+                % (
+                    cookie_name,
+                    cookie.encode("utf-8"),
+                    self._callback_path_prefix.encode("utf-8")
+                    + b"/_synapse/client/oidc",
```

Comment:
why not include `/_synapse/client/oidc` in `self._callback_path` ?

Reply:
Hmm, I suppose it's handy to have a reference to the path extracted from `public_baseurl`, which we can then add on to in places (as we do here).

But while we're only using this in one place, we can move the concatenation out of the weeds/runtime code path.